### PR TITLE
use test -L and -e to judge whether file exists.

### PR DESCRIPTION
**Pull request by detailyang**
when symbolic link target_file has moved test -e $path will return false for issues #182

### CODE REVIEW

**Comment by bucaran** [comment]
@xrain0610 Can you check if this fixes your use case?

Also, @detailyang wrap those in "$variable" double quotes. If the var has no value it will expand to nothing causing the test builtin to break or act unexpectedly.

@jethrokuan Ping.

**Comment by detailyang** [comment]
@bucaran what's the meaning `wrap those in "$variable" double quotes`?  i havent understand "If the var has no value it will expand to nothing causing the test builtin to break or act unexpectedly."
sorry for this, any more detailed example for me? :smile:

File: functions/__fisher_plugin_fetch.fish [hunk 1]
```diff
@@ -72,6 +72,14 @@ function -S __fisher_plugin_fetch
         debug "plugin %s" "$name"
 
         if test ! -e $path
+            if test -L $path
```
Comment (bucaran) [inline functions/__fisher_plugin_fetch.fish]: ```test -L "$path"```

File: functions/__fisher_plugin_fetch.fish [hunk 2]
```diff
@@ -72,6 +72,14 @@ function -S __fisher_plugin_fetch
         debug "plugin %s" "$name"
 
         if test ! -e $path
```
Comment (bucaran) [inline functions/__fisher_plugin_fetch.fish]: ```test ! -e "$path"```

File: functions/__fisher_plugin_fetch.fish [hunk 3]
```diff
@@ -72,6 +72,14 @@ function -S __fisher_plugin_fetch
...
+            if test -L $path
+                set -l realpath (command readlink $path)
```
Comment (bucaran) [inline functions/__fisher_plugin_fetch.fish]: ```command readlink "$path"```

**Comment by bucaran** [comment]
Sure, I left you some comments.

**Comment by detailyang** [comment]
thx, i know what you mean:)

**Comment by bucaran** [comment]
OK @detailyang are we ready to merge?

**Comment by detailyang** [comment]
@bucaran  yep, go aheah

**Comment by bucaran** [comment]
@detailyang Thanks! Let's how it goes :smile:

**Comment by detailyang** [comment]
haha, first merged PR to fisherman :grin:

**Comment by bucaran** [comment]
Keep 'em coming!

**Pull request merged**